VmAgent.Core/Interfaces/ProcessRunner.cs

```diff
@@ -58,8 +69,10 @@ public override Task<SessionHostInfo> CreateAndStart(int instanceNumber, GameRes
 
             try
             {
-                string processId = _processWrapper.Start(processStartInfo).ToString();
-                sessionHostManager.UpdateSessionHostTypeSpecificId(sessionHostUniqueId, processId);
+                int processId = processOutputLogger != null ? 
+                    _processWrapper.StartWithEventHandler(processStartInfo, processOutputLogger.StdOutputHandler, processOutputLogger.ErrorOutputHandler) : _processWrapper.Start(processStartInfo);
```

Comment:
Both stdout and stderr are being written to the same file here, which might be fine but i think it would be better if they wrote to separate files. Currently there is no way to tell what is written to each stream--they both just look like text in the output file (in a cmd window you can tell because stderr text is usually written in a different color).
Also i don't know what considerations need to be taken about multithreading--if the game server writes to stdout and stderr at the same time, will the text get jarbled in the output file? Or will something worse happen like a crash on our end? If we have them write to separate files we avoid having to worry about all this

Reply:
I think we want to keep the same log file as Container-based Game Server.  I like the idea that we separate the log files for different purposes but we might need to do the same thing for Container? I'm going through documents if writing a log to a same file from two different event handler would cause a problem.

Reply:
If we want them in the same file, we should do something like what Docker does for their console output logs. Here's an example:
```
{"log":"Hello, World! \u003c- written to console before GSDK initialized\n","stream":"stdout","time":"2022-08-17T17:54:26.185099166Z"}
{"log":"{\n","stream":"stdout","time":"2022-08-17T17:54:26.305422455Z"}
{"log":"  \"certificateFolder\": \"/data/GameCertificates\",\n","stream":"stdout","time":"2022-08-17T17:54:26.305445455Z"}
{"log":"  \"buildId\": \"bc452904-c2f1-4acd-b068-4ddddfd1a34c\",\n","stream":"stdout","time":"2022-08-17T17:54:26.305451955Z"}
{"log":"  \"logFolder\": \"/data/GameLogs/\",\n","stream":"stdout","time":"2022-08-17T17:54:26.305456255Z"}
{"log":"  \"vmId\": \"vmss:EastUs:59F84_bdfab727-1382-47a0-be31-a960577004d0:c7ae0ca6-38bc-4774-805e-5bec765e6cdb\",\n","stream":"stdout","time":"2022-08-17T17:54:26.305460155Z"}
```
Each of these lines specifies that it was written to the "stdout" steam (and presumably would specify if something was written to the "stderr" stream as well). And if we're adding additional data to each line, we might as well add the datetime too, because that's probably useful as well

Reply:
Thanks Brian, yes this is definitely more useful - I'll do research on how two ( or three if we add exit handler) event handler would write a file each time and share what approach would be the best